model/DataObject.php

```diff
@@ -1647,6 +1648,121 @@ public function getRelationClass($relationName) {
 	}
 
 	/**
+	 * Given a relation name, determine the relation type
+	 *
+	 * @param string $component Name of component
+	 * @return string has_one, has_many, many_many, belongs_many_many or belongs_to
+	 */
+	public function getRelationType($component) {
+		$types = array('has_one', 'has_many', 'many_many', 'belongs_many_many', 'belongs_to');
+		foreach($types as $type) {
+			$relations = Config::inst()->get($this->class, $type);
+			if($relations && isset($relations[$component])) {
+				return $type;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Given a relation declared on a remote class, mock the component.
+	 *
+	 * Notes on behaviour:
+	 *  - This can still be used on components that are defined on both sides, but do not need to be.
+	 *  - All has_ones on remote class will be treated as local has_many, even if they are belongs_to
+	 *  - Cannot be used on polymorphic relationships
+	 *  - Cannot be used on unsaved objects.
+	 *
+	 * @param string $remoteClass
+	 * @param string $remoteRelation
+	 * @return DataList|DataObject The component, either as a list or single object
+	 * @throws BadMethodCallException
+	 * @throws InvalidArgumentException
+	 */
+	public function getNonReciprocalComponent($remoteClass, $remoteRelation) {
```

Comment:
I think I should use 'reciprocative' not 'reciprocal' though.... it might be better grammar. :)